### PR TITLE
Minimize the race condition in BackoffSweeper()

### PR DESCRIPTION
**Pull request by pengzhout**
There is a long-standing race condition in BackoffSweeper() which
triggers an error and then triggers another assertion failure for
not reset sweeperInProgress to false.

This commit doesn't resolve the race condition fundamentally with
lock or other implementation, because the whole backoff mechanism
did not ask for accurate control, so skipping some sweeps should
be fine so far. We also downgrade the log level to DEBUG because
a restart of sweeper backend is unnecessary.

this PR can resolve the intermittent failure in pipeline job icw_planner_sles12.

### CODE REVIEW

**Comment by hlinnaka** [comment]
Thanks for addressing this! It's been an annoying failure.

File: src/backend/postmaster/backoff.c [hunk 1]
```diff
@@ -857,12 +863,30 @@ BackoffSweeper()
...
+						 * Timestamp4: Sweeper try to find the backends who deserve maxCPU,
+						 * if backend A, B, C all deserve maxCPU, then activeWeight = 
+						 * 200000 - 100000/1 - 100000/1 - 100000/2 which is less than zero.
+						 *
+						 * We can stop sweeping for such race condition because current
+						 * backoff mechanism dose not ask for accurate control.
 						 */
-						elog(ERROR, "activeWeight underflow!");
+						backoffSingleton->sweeperInProgress = false;
+						elog(DEBUG1, "activeWeight underflow!");
```
Comment (hsyuan) [inline src/backend/postmaster/backoff.c]: maybe better to use "LOG". It was LOG before we changed it to ERROR: https://github.com/greenplum-db/gpdb/commit/c7befb1d1b64f1e646a1eecb78e5fdb864717471

Comment (pengzhout) [reply src/backend/postmaster/backoff.c]: OK, thanks

**Comment by pengzhout** [comment]
merged

**Pull request merged**